### PR TITLE
Reset signals in Authz helper

### PR DESCRIPTION
**Pull request by HereThereBeDragons**
Issue https://github.com/cvmfs/cvmfs/issues/3211

Supersedes https://github.com/cvmfs/cvmfs/pull/3256

Same as PR #3260, just wrong branch name... and did not know that renaming a branch on your fork just closes the PR... (no warning or mentioning was given)

### CODE REVIEW

**Comment by HereThereBeDragons** [comment]
@cernvm-bot cloudtest

**Comment by HereThereBeDragons** [comment]
@cernvm-bot cloudtest

**Comment by HereThereBeDragons** [comment]
`test 811` fails sometimes with errno 10: No child processes 
in 
 https://github.com/cvmfs/cvmfs/blob/5d9a676aefc000b22786d429ee0bb60da938d5c5/cvmfs/publish/cmd_enter.cc#L671
 
 should it be there more tolerant? and also be a success if the child is already dead?

File: cvmfs/monitor.h [hunk 1]
```diff
@@ -12,6 +12,7 @@
 
 #include <map>
 #include <string>
+#include <vector>
```
Comment (jblomer) [inline cvmfs/monitor.h]: Not needed

File: cvmfs/monitor.h [hunk 2]
```diff
@@ -42,6 +43,15 @@ class Watchdog : SingleCopy {
   ~Watchdog();
   void Spawn(const std::string &crash_dump_path);
 
+  /**
+   * All signals overwritten by the
```
Comment (jblomer) [inline cvmfs/monitor.h]:
```suggestion
   * All signals overwritten by the monitor
```

File: cvmfs/monitor.h [hunk 3]
```diff
@@ -42,6 +43,15 @@ class Watchdog : SingleCopy {
...
+  */
+  static int all_signals[13];
```
Comment (jblomer) [inline cvmfs/monitor.h]: Indicate as a global:
```suggestion
  static int g_all_signals[13];
```

File: cvmfs/monitor.h [hunk 4]
```diff
@@ -42,6 +43,15 @@ class Watchdog : SingleCopy {
...
+  /**
+   * Signals used by crash signal handler
+  */
+  static int crash_signals[8];
```
Comment (jblomer) [inline cvmfs/monitor.h]:
```suggestion
  static int g_crash_signals[8];
```

**Review by jblomer — CHANGES_REQUESTED** [review]
LGTM, just a few minor comments

File: cvmfs/monitor.h (hunk 3 above)
Comment (HereThereBeDragons) [reply cvmfs/monitor.h]: // The watchdog is not supposed to receive signals. If it does, report it.

I therefore suggest renaming it to `g_suppressed_signals`

**Comment by HereThereBeDragons** [comment]
@cernvm-bot cloudtest

**Comment by HereThereBeDragons** [comment]
@cernvm-bot cloudtest

**Review by jblomer — APPROVED** [review]

**Comment by HereThereBeDragons** [comment]
@cernvm-bot cloudtest

**Review by vvolkl — APPROVED** [review]

**Pull request merged**